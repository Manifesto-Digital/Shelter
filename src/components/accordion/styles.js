import styled from 'styled-components';
import SVG from 'react-inlinesvg';
import { buttonReset } from '../styled/buttons';

export const Wrapper = styled.div`
  position: relative;

  &:not(:first-of-type) {
    &:before {
      position: absolute;
      content: ' ';
      left: 0;
      top: ${({ theme }) => `-${theme.spacing.small}`};
      width: 100%;
      border-top: ${({ theme, active }) =>
        active && `1px solid ${theme.palette.grey45}`};
    }
  }
`;

export const Heading = styled.h3`
  position: relative;
  font-weight: normal;
  color: ${({ theme }) => theme.palette.grey10};
  margin-bottom: ${({ active }) => active && '0'};
`;

export const HeadingButton = styled.button`
  position: relative;
  ${buttonReset};
  padding: ${({ theme }) =>
    `${theme.spacing.standard} ${theme.spacing.xl} ${theme.spacing.standard} ${
      theme.spacing.standard
    }`};
  width: 100%;
  text-align: left;
  font-size: ${({ theme }) => theme.headers.h3};
`;

export const Content = styled.div`
  display: ${({ isOpen }) => (isOpen ? 'block' : 'none')};
  padding: ${({ theme }) =>
    `${theme.spacing.standard} ${theme.spacing.standard} 0 ${
      theme.spacing.standard
    }`};
`;

export const ArrowSVG = styled(SVG)`
  position: absolute;
  transform: ${({ isOpen }) => isOpen && 'rotate(180deg)'} translateY(-50%);
  transform-origin: top;
  top: 50%;
  right: ${({ theme }) => theme.spacing.standard};
  width: 20px;
  height: 25px;

  svg {
    display: block;
    width: 20px;
    height: 25px;
  }
`;
