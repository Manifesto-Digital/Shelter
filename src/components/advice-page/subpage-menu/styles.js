import styled from 'styled-components';

export const MenuNav = styled.nav`
  margin-bottom: ${({ theme }) => theme.spacing.large};

  a {
    text-decoration: none;
  }
`;
