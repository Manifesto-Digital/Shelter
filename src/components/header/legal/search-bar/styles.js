import styled from 'styled-components';
import SVG from 'react-inlinesvg';
import { breakpoint } from '../../../theme/breakpoint';
import { buttonReset } from '../../../styled/buttons';
import { inputStyles } from '../../../styled/inputs';

export const FormWrapper = styled.form`
  display: none;
  height: 43px;

  ${breakpoint.desktop`
      display: flex;
  `}
`;

export const InputWrap = styled.div`
  display: flex;
  position: relative;
  margin-right: 5px;
  font-size: ${({ theme }) => theme.fontsize.small};
  width: 100%;
  margin: 0 ${({ theme }) => theme.spacing.small};
  border: 1px solid ${({ theme }) => theme.palette.grey10};

  ${breakpoint.desktop`
    width: auto;
    border: 0;
  `}
`;

export const SearchInput = styled.input`
  ${inputStyles}
  width: 100%;
  margin-right: 5px;
  background-color: ${({ theme }) => theme.palette.white};
  border: 1px solid ${({ theme }) => theme.palette.black};
  padding-left: ${({ theme }) => theme.spacing.large};

  ${breakpoint.desktop`
      min-width: 320px;
  `}
`;

export const SearchIcon = styled(SVG)`
  display: block;
  position: absolute;
  left: 15px;
  width: 18px;
  top: 50%;
  transform: translateY(-50%);
  fill: ${({ theme }) => theme.palette.grey66};

  svg {
    display: block;
  }
`;

export const SearchButton = styled.button`
  ${buttonReset}
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 0 18px;
  color: ${({ theme }) => theme.palette.white};
  background-color: ${({ theme }) => theme.palette.grey80};
`;
