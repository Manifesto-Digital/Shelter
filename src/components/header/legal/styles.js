import styled from 'styled-components';
import { breakpoint } from '../../theme/breakpoint';
import { buttonStyles, buttonReset } from '../../styled/buttons';
import LinkHandler from '../../link-handler';

export const LegalDonateButton = styled(LinkHandler)`
  ${buttonStyles};
  display: flex;
  align-items: center;
  justify-content: center;
  margin-right: ${({ mobileMenu, theme }) => mobileMenu && theme.spacing.small};
  text-decoration: none;

  ${breakpoint.desktop`
    height: 100%;
  `};

  &:hover {
    color: ${({ theme }) => theme.palette.white};
  }
`;

export const LegalMenuList = styled.ul`
  display: flex;
  width: 100%;
  list-style: none;
  margin: 0;
  padding: 0;
`;

export const SecondHeaderWrapper = styled.div`
  display: none;
  padding: ${({ theme }) => theme.spacing.small} 0;
  background-color: ${({ theme }) => theme.palette.grey25};
  ${breakpoint.desktop`
    display: block;
  `};
`;

export const SecondHeaderBar = styled.div`
  display: flex;
  align-items: center;
  margin-bottom: ${({ theme }) => theme.spacing.small};
`;

export const SubsectionHeader = styled.h3`
  margin-right: ${({ theme }) => theme.spacing.large};
  margin-bottom: 0;
  font-weight: normal;
`;

export const LegalItem = styled.li`
  display: flex;
  align-items: center;
  position: relative;
  margin: 0 ${({ theme }) => theme.spacing.standard} 0 0;
  font-size: ${({ theme }) => theme.fontsize.small};
`;

export const MoreItemMenu = styled.button`
  ${buttonReset};
  font-size: ${({ theme }) => theme.fontsize.small};
`;

export const ThirdNavigationBar = styled.div`
  display: flex;
  background-color: ${({ theme }) => theme.palette.grey25};
  padding: ${({ theme }) => theme.spacing.small} 0;

  a {
    font-weight: normal;
  }
`;

/* Currently only display on desktop */
export const MoreSubMenu = styled.ul`
  position: absolute;
  display: flex;
  flex-direction: column;
  top: calc(100% + ${({ theme }) => theme.spacing.standard});
  right: 0;
  margin: 0;
  min-width: 225px;
  padding: ${({ theme }) => theme.spacing.standard};
  border: none;
  list-style: none;
  background: ${({ theme }) => theme.palette.grey15};

  ${LegalItem} {
    margin-bottom: ${({ theme }) => theme.spacing.standard};
    &:last-child {
      border: none;
      margin-bottom: 0;
    }
  }
`;
