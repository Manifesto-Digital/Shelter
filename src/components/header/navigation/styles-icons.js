import styled from 'styled-components';
import SVG from 'react-inlinesvg';
import { buttonReset } from '../../styled/buttons';
import { breakpoint } from '../../theme/breakpoint';

export const MobileMenuClose = styled.button`
  ${buttonReset}
  display: flex;
  justify-content: center;
  align-items: center;
  width: 45px;
  height: 40px;
  padding: 0;
  margin-left: auto;
  cursor: pointer;

  ${breakpoint.desktop`
    display: none;
  `}
`;

export const SubNavButton = styled(MobileMenuClose)`
  height: auto;
  width: 25px;

  ${breakpoint.desktop`
    display: ${({ legal }) => (legal ? 'none' : 'flex')};
    margin-left: 5px;

    svg {
      width: auto;
    }
  `}

  svg {
    width: 15px;
  }
`;

export const MenuSVG = styled(SVG)`
  display: block;
  width: 20px;
`;

export const ArrowSVG = styled(SVG)`
  display: block;
  fill: ${({ theme }) => theme.palette.black};
  width: 20px;
  cursor: pointer;
  pointer-events: none;
`;
