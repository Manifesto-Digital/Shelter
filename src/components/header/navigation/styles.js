import styled from 'styled-components';
import { breakpoint } from '../../theme/breakpoint';
import LinkHandler from '../../link-handler';

export const Wrapper = styled.div`
  position: fixed;
  top: 0;
  right: 0;
  bottom: 0;
  width: 80%;
  height: 100%;
  background: ${({ theme }) => theme.palette.white};
  box-shadow: ${({ theme }) => theme.boxshadow.menu};
  transition: transform 0.3s cubic-bezier(0, 0, 0.3, 1);
  transform: ${({ active }) =>
    active ? 'translateX(0)' : ' translateX(120%)'};
  z-index: ${({ active }) => (active ? '10' : 1)};
  overflow-y: scroll;

  ${breakpoint.desktop`
    position: relative;
    width:100%;
    background: ${({ theme }) => theme.palette.grey10};
    box-shadow: none;
    transition: none;
    transform: translateX(0);
    overflow-y: visible;
  `}
`;

export const MenuList = styled.ul`
  display: flex;
  flex-direction: column;
  width: 100%;
  list-style: none;
  margin: 0;
  padding: 0;

  ${breakpoint.desktop`
    flex-direction: row;
    width: auto;
  `}
`;

export const Menus = styled.nav`
  position: relative;
  display: flex;
  flex-direction: column;
  width: 100%;
  height: 100%;
  padding: 0;

  ${breakpoint.desktop`
    flex-direction: row;
  justify-content: ${({ legal }) => legal && 'flex-end'};
    max-width: ${({ theme }) => theme.sizes.maxWidth};
    margin: 0 auto;
    padding: 0 ${({ theme }) => theme.spacing.small};
  `}
`;

export const Item = styled.li`
  display: flex;
  align-items: center;
  margin-bottom: 0;
  padding: ${({ topLevel, legal, theme }) =>
    (legal && topLevel && `15px ${theme.spacing.standard}`) ||
    (topLevel && '0') ||
    `${theme.spacing.small}}`};
  border-bottom: ${({ topLevel, theme }) =>
    topLevel ? ' none' : `1px solid ${theme.palette.grey10}`};
  border-bottom: ${({ topLevel, legal, theme }) =>
    (legal && topLevel && `1px solid ${theme.palette.grey10}`) ||
    (topLevel && 'none') ||
    `1px solid ${theme.palette.grey10}`};

  ${MenuList} {
    padding: 5px ${({ theme }) => theme.spacing.small};
    ${breakpoint.desktop`
      padding:5px 0;
    `}

    ${breakpoint.desktop`
    margin-right: auto;
     padding: ${({ topLevel, theme }) =>
       topLevel ? `0` : `${theme.spacing.small}`};
  `}
  }
`;

export const ItemLink = styled(LinkHandler)`
  color: ${({ theme }) => theme.palette.black};
  font-size: ${({ theme }) => theme.fontsize.small};
  text-decoration: none;
`;

export const AdditionalMenu = styled(MenuList)`
  padding: 0 ${({ theme }) => theme.spacing.small};
  ${Item} {
    padding: 15px ${({ theme }) => theme.spacing.small};
    &:last-child {
      border: none;
      ${breakpoint.desktop`
        margin-left: ${({ theme }) => theme.spacing.standard};
      `}
    }
  }
  ${breakpoint.desktop`
    flex-direction: row;
    min-width: 200px;
    margin-left: auto;
    margin-right: 0;
      ${ItemLink} {
        border:none;
      }
  `}
`;

export const SubMenu = styled.ul`
  display: ${({ active }) => (active ? 'flex' : 'none')};
  flex-direction: column;
  padding: 0 ${({ theme }) => theme.spacing.small};
  border: none;
  list-style: none;

  ${breakpoint.desktop`
    position: absolute;
    display: ${({ legal }) => legal && 'none'};
    top: 52px;
    right: 0;
    left: 0;
    flex-direction: row;
    flex-wrap: wrap;
    width: 100%;
    height: auto;
    margin: 0;
    background: ${({ theme }) => theme.palette.grey15};
    padding: ${({ theme }) => theme.spacing.small} ${({ theme }) =>
    theme.spacing.small};
  `}

  ${Item} {
    &:last-child {
      border: none;
    }
    ${breakpoint.desktop`
      flex: 0 1 calc(${100 / 3}% - 10px);
      width: 32%;
      margin-left: ${({ theme }) => theme.spacing.small};
      border: none;
    `}

    &:hover {
      ${breakpoint.desktop`
          background: ${({ theme }) => theme.palette.black};
        ${ItemLink} {
          color: ${({ theme }) => theme.palette.white};
        }
      `}
    }
  }
`;

export const SkipToContent = styled.a`
  display: block;
  width: 100%;
  max-width: ${({ theme }) => theme.sizes.maxWidth};
  height: 0;
  margin: 0 auto;
  overflow: hidden;

  &:focus {
    height: auto;
  }
`;
