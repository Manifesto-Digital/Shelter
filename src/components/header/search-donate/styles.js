import styled from 'styled-components';
import SVG from 'react-inlinesvg';
import { breakpoint } from '../../theme/breakpoint';
import { buttonReset, buttonStyles } from '../../styled/buttons';
import { inputStyles } from '../../styled/inputs';
import LinkHandler from '../../link-handler';

import theme from '../../theme/variables';

export const InputWrap = styled.div`
  display: flex;
  margin-right: ${theme.spacing.xs};
  font-size: ${theme.fontsize.small};
  width: 100%;
  margin: 0 ${theme.spacing.small};
  border: 1px solid ${theme.palette.grey10};

  ${breakpoint.desktop`
    width: auto;
    border: 0;
  `}
`;

export const SearchInput = styled.input`
  ${inputStyles}
  border: 0;
  width: 100%;

  ${breakpoint.desktop`
      min-width: 185px;
  `}
`;

export const SearchButton = styled.button`
  ${buttonReset}
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 0 ${theme.spacing.small};
  background-color: ${theme.palette.white};
`;

export const SearchIcon = styled(SVG)`
  display: block;
  width: 18px;
  fill: ${theme.palette.royalBlue};
`;

export const DonateButton = styled(LinkHandler)`
  ${buttonStyles};
  display: flex;
  align-items: center;
  justify-content: center;
  margin-right: ${({ mobileOnly }) => mobileOnly && theme.spacing.small};
  text-decoration: none;
  ${breakpoint.desktop`
    display: ${({ mobileOnly }) => (mobileOnly ? 'none' : 'flex')};
  `}

  &:hover {
    color: ${theme.palette.white};
  }
`;

export const Wrapper = styled.form`
  display: ${({ resolution }) => (resolution === 'desktop' ? 'none' : 'flex')};

  ${breakpoint.desktop`
      display: ${({ resolution }) =>
        resolution === 'desktop' ? 'flex' : 'none'};
  `}

  ${DonateButton} {
    display: ${({ resolution }) => (resolution === 'mobile' ? 'none' : 'flex')};
  }

  height: 43px;
`;
