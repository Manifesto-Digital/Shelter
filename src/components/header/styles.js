import styled from 'styled-components';
import SVG from 'react-inlinesvg';
import { buttonReset } from '../styled/buttons';
import { breakpoint } from '../theme/breakpoint';

export const Logo = styled(SVG)`
  display: block;
  width: 95px;

  ${breakpoint.desktop`
    width: 115px;
  `}
`;

export const LogoWrapper = styled.a`
  display: flex;
  align-items: center;
  justify-content: center;
  height: 52px;
  padding: 15px;
  background: ${({ theme }) => theme.palette.primary};

  ${breakpoint.desktop`
    height: 68px;
    padding: ${({ theme }) => theme.spacing.standard} 25px;
  `}
`;

export const Open = styled.button`
  ${buttonReset}
  display: flex;
  justify-content: center;
  width: 45px;
  height: 50px;
  padding: 0;
  margin-left: auto;

  ${breakpoint.desktop`
    display: none;
  `}
`;

export const BurgerIcon = styled(SVG)`
  display: block;
  width: 20px;
`;

export const HeaderWrapper = styled.header`
  display: flex;
  justify-content: center;
  width: 100%;
  background: ${({ theme }) => theme.palette.grey10};
`;

export const HeaderBar = styled.header`
  display: flex;
  justify-content: space-between;
  max-width: 1150px;
  width: 100%;
  background: ${({ theme }) => theme.palette.grey10};

  ${breakpoint.desktop`
    padding: 0
    ${({ theme }) => theme.spacing.standard};
  `}
`;
