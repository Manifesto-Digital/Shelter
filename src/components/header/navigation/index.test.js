import React from 'react';
import 'jest-styled-components';
import { snapshotComponent, mountWithTheme } from 'test-helpers';
import {
  createHeaderNavigation,
  createInternalRef,
} from '../../../utils/test-factories';
import Navigation from '.';
import NavigationMenu from './navigation';
import { MobileMenuClose, SubNavButton } from './styles-icons';
import { ItemLink, SubMenu } from './styles';
import { hidePascalCaseWarning } from '../../../utils/test-mocks';

it('Renders correctly', () => {
  const mockData = createHeaderNavigation();
  snapshotComponent(<Navigation pageData={mockData} active />);
});

hidePascalCaseWarning();

it('Displays close button', () => {
  const mockData = createHeaderNavigation();
  const wrapper = mountWithTheme(<Navigation pageData={mockData} active />);
  expect(wrapper.find(MobileMenuClose)).toHaveLength(1);
});

test('Should populate a menu item correctly', () => {
  const mockData = createHeaderNavigation({
    id: 'e230d8b8-4ee6-5d4c-bf25-57af664d12d7',
    menuLabel: 'What we do',
    navigationLink: [createInternalRef({ title: 'title' })],
  });
  const wrapper = mountWithTheme(
    <NavigationMenu pageData={mockData} id={mockData.id} />
  );

  expect(
    wrapper
      .find(ItemLink)
      .at(0)
      .text()
  ).toBe(mockData.menuLabel || mockData.navigationLink[0].title);

  expect(
    wrapper
      .find(ItemLink)
      .at(0)
      .props().internalLink
  ).toBe(mockData.navigationLink[0]);
});

test('Should populate a sub-menu', () => {
  const mockData = createHeaderNavigation({
    id: 'e230d8b8-4ee6-5d4c-bf25-57af664d12d7',
    subNavigationItems: [
      {
        slug: '320,000-people-in-britain-are-now-homeless',
        title: '320,000 people in Britain are now homeless',
      },
      {
        title: 'Shelter Manchester to hold vigil for lost homeless people',
        slug: 'shelter-manchester-to-hold-vigil-for-lost-homeless-people',
      },
    ],
  });
  const wrapper = mountWithTheme(
    <NavigationMenu pageData={mockData} id={mockData.id} />
  );

  expect(wrapper.find(ItemLink)).toHaveLength(
    mockData.subNavigationItems.length
  );
});

it('Opens a sub-navigation on button click', () => {
  const mockData = createHeaderNavigation({
    navigationItems: [
      {
        id: 'd582da1f-1f7f-5f4c-ae3b-ef2b41972dbc',
        menuLabel: 'Housing Advice',
        navigationLink: [{ title: 'Tobys Page', slug: 'tobys-page' }],
        subNavigationItems: [
          {
            slug: '320,000-people-in-britain-are-now-homeless',
            title: '320,000 people in Britain are now homeless',
          },
          {
            title: 'Tobys Page',
            slug: 'tobys-page',
          },
          {
            title: 'Shelter Manchester to hold vigil for lost homeless people',
            slug: 'shelter-manchester-to-hold-vigil-for-lost-homeless-people',
          },
          {
            title: 'Shelter Demo Page',
            slug: 'shelter-demo-page',
          },
        ],
      },
    ],
  });

  const wrapper = mountWithTheme(<Navigation pageData={mockData} active />);
  const mockButton = wrapper.find(SubNavButton);

  mockButton.simulate('click');
  expect(wrapper.find(SubMenu)).toHaveStyleRule('display', 'flex');
});
