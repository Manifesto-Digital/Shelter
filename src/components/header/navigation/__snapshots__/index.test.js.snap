// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
.c1 {
  padding: 0;
  border: none;
  font: inherit;
  color: inherit;
  background-color: transparent;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 45px;
  height: 40px;
  padding: 0;
  margin-left: auto;
  cursor: pointer;
}

.c11 {
  padding: 0;
  border: none;
  font: inherit;
  color: inherit;
  background-color: transparent;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 45px;
  height: 40px;
  padding: 0;
  margin-left: auto;
  cursor: pointer;
  height: auto;
  width: 25px;
  margin-left: 5px;
}

.c11 svg {
  width: 15px;
}

.c2 {
  display: block;
  width: 20px;
}

.c12 {
  display: block;
  fill: #212122;
  width: 20px;
  cursor: pointer;
}

.c0 {
  position: fixed;
  top: 0;
  right: 0;
  bottom: 0;
  z-index: 10;
  width: 80%;
  height: 100%;
  background: #fff;
  box-shadow: -1px 0 14px rgba(0,0,0,0.5);
  -webkit-transition: -webkit-transform 0.3s cubic-bezier(0,0,0.3,1);
  -webkit-transition: transform 0.3s cubic-bezier(0,0,0.3,1);
  transition: transform 0.3s cubic-bezier(0,0,0.3,1);
  -webkit-transform: translateX(0);
  -ms-transform: translateX(0);
  transform: translateX(0);
  overflow-y: scroll;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  list-style: none;
  margin: 0;
  padding: 0;
}

.c3 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  padding: 0;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-bottom: 0;
  padding: 0;
  border-bottom: none;
}

.c7 .c4 {
  padding: 5px 10px;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-bottom: 0;
  padding: 10px;
  border-bottom: 1px solid #e5e5e5;
}

.c8 .c4 {
  padding: 5px 10px;
}

.c10 {
  color: #212122;
  font-size: 0.775rem;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c14 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  list-style: none;
  margin: 0;
  padding: 0;
  padding: 0 10px;
}

.c14 .c6 {
  padding: 15px 10px;
}

.c14 .c6:last-child {
  border: none;
}

.c13 {
  display: none;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 0 10px;
  border: none;
  list-style: none;
}

.c13 .c6:last-child {
  border: none;
}

@media (min-width:62em) {
  .c1 {
    display: none;
  }
}

@media (min-width:62em) {
  .c11 {
    display: none;
  }
}

@media (min-width:62em) {
  .c11 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }

  .c11 svg {
    width: auto;
  }
}

@media (min-width:62em) {
  .c0 {
    position: relative;
    width: 100%;
    background: #e5e5e5;
    box-shadow: none;
    -webkit-transition: none;
    transition: none;
    -webkit-transform: translateX(0);
    -ms-transform: translateX(0);
    transform: translateX(0);
    overflow-y: visible;
  }
}

@media (min-width:62em) {
  .c5 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    width: auto;
    margin-right: 20px;
  }
}

@media (min-width:62em) {
  .c3 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    max-width: 1140px;
    margin: 0 auto;
    padding: 0 10px;
  }
}

@media (min-width:62em) {
  .c7 {
    margin-right: auto;
    padding: 0;
  }
}

@media (min-width:62em) {
  .c8 {
    margin-right: auto;
    padding: 10px;
  }
}

@media (min-width:62em) {
  .c14 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    width: auto;
    margin-right: 20px;
  }
}

@media (min-width:62em) {
  .c14 .c6:last-child {
    margin-left: 20px;
  }
}

@media (min-width:62em) {
  .c14 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    min-width: 200px;
    margin-left: auto;
    margin-right: 0;
  }

  .c14 .c9 {
    border: none;
  }
}

@media (min-width:62em) {
  .c13 {
    position: absolute;
    top: 58px;
    right: 0;
    left: 0;
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    -webkit-flex-wrap: wrap;
    -ms-flex-wrap: wrap;
    flex-wrap: wrap;
    width: 100%;
    height: auto;
    margin: 0;
    background: #d9d9d9;
    padding: 10px 10px;
  }
}

@media (min-width:62em) {
  .c13 .c6 {
    -webkit-flex: 0 1 calc(33.333333333333336% - 10px);
    -ms-flex: 0 1 calc(33.333333333333336% - 10px);
    flex: 0 1 calc(33.333333333333336% - 10px);
    width: 32%;
    margin-left: 10px;
    border: none;
  }
}

@media (min-width:62em) {
  .c13 .c6:hover {
    background: #212122;
  }

  .c13 .c6:hover .c9 {
    color: #fff;
  }
}

<div
  className="c0"
>
  <button
    className="c1"
    type="button"
  >
    <span
      className="isvg loading c2"
    />
  </button>
  <nav
    aria-label="Main menu"
    className="c3"
    role="navigation"
  >
    <ul
      aria-hidden="false"
      className="c4 c5"
      role="menubar"
    >
      <li
        className="c6 c7"
      >
        <ul
          className="c4 c5"
        >
          <li
            aria-haspopup={true}
            className="c6 c8"
            role="menuitem"
          >
            <Link
              className="c9 c10"
              to="tobys-page"
            >
              Housing Advice
            </Link>
            <button
              className="c11"
              onClick={[Function]}
              type="button"
            >
              <span
                className="isvg loading c12"
              />
            </button>
          </li>
          <ul
            aria-hidden="true"
            className="c13"
            role="menu"
          >
            <li
              className="c6 c8"
            >
              <Link
                className="c9 c10"
                to="320,000-people-in-britain-are-now-homeless"
              >
                320,000 people in Britain are now homeless
              </Link>
            </li>
            <li
              className="c6 c8"
            >
              <Link
                className="c9 c10"
                to="tobys-page"
              >
                Tobys Page
              </Link>
            </li>
            <li
              className="c6 c8"
            >
              <Link
                className="c9 c10"
                to="shelter-manchester-to-hold-vigil-for-lost-homeless-people"
              >
                Shelter Manchester to hold vigil for lost homeless people
              </Link>
            </li>
            <li
              className="c6 c8"
            >
              <Link
                className="c9 c10"
                to="shelter-demo-page"
              >
                Shelter Demo Page
              </Link>
            </li>
          </ul>
        </ul>
      </li>
    </ul>
    <ul
      aria-label="Secondary menu"
      className="c4 c14"
      role="navigation"
    >
      <li
        className="c6 c8"
      >
        <Link
          className="c9 c10"
          to="tobys-page"
        >
          Tobys Page
        </Link>
      </li>
      <li
        className="c6 c8"
      >
        <Link
          className="c9 c10"
        >
          Scotland
        </Link>
      </li>
    </ul>
  </nav>
</div>
`;
